### PR TITLE
Re-enable Ballerina metrics test cases

### PR DESCRIPTION
**Pull request by nadundesilva**
## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/observability/metrics/MetricsTestCase.java [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+            }
+        });
+
+        Assert.assertEquals(functionMetrics.getGauges().stream()
+                        .map(gauge -> gauge.getId().getName())
+                        .collect(Collectors.toSet()),
+                new HashSet<>(Arrays.asList("response_time_seconds", "inprogress_requests")));
+        Assert.assertEquals(functionMetrics.getGauges().size(), 2);
+        functionMetrics.getGauges().forEach(gauge -> {
+            if (Objects.equals(gauge.getId().getName(), "response_time_seconds")) {
```
Comment (Kishanthan) [inline tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/observability/metrics/MetricsTestCase.java:127]: This if condition block seems to be too big and should be refactored into multiple methods for better readability.

File: tests/observability-test-utils/src/main/resources/META-INF/services/org.ballerinalang.jvm.observability.metrics.spi.MetricReporter [hunk 2]
```diff
@@ -0,0 +1 @@
+org.ballerina.testobserve.metrics.extension.BMockMetricsReporter
```
Comment (Kishanthan) [inline tests/observability-test-utils/src/main/resources/META-INF/services/org.ballerinalang.jvm.observability.metrics.spi.MetricReporter:1]: add EoF line here.

File: tests/observability-test-utils/src/main/java/org/ballerina/testobserve/metrics/extension/MetricsUtils.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+                mockGauge.setCount(gauge.getCount());
+                mockGauge.setSum(gauge.getSum());
+                mockGauge.setSnapshots(gauge.getSnapshots());
+                metrics.addGauge(mockGauge);
+            } else if (metric instanceof PolledGauge) {
+                PolledGauge polledGauge = (PolledGauge) metric;
+                MockPolledGauge mockPolledGauge = new MockPolledGauge();
+                mockPolledGauge.setId(polledGauge.getId());
+                mockPolledGauge.setValue(polledGauge.getValue());
+                metrics.addPolledGauge(mockPolledGauge);
```
Comment (Kishanthan) [inline tests/observability-test-utils/src/main/java/org/ballerina/testobserve/metrics/extension/MetricsUtils.java:61]: Only way is to check instanceof here? Can't we have some type / enum based cases?

File: tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/observability/metrics/MetricsTestCase.java (hunk 1 above)
Comment (nadundesilva) [reply tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/observability/metrics/MetricsTestCase.java:184]: Fixed in f2a17a0d2edd186d40174bd5bea21f9551b363a6

File: tests/observability-test-utils/src/main/resources/META-INF/services/org.ballerinalang.jvm.observability.metrics.spi.MetricReporter (hunk 2 above)
Comment (nadundesilva) [reply tests/observability-test-utils/src/main/resources/META-INF/services/org.ballerinalang.jvm.observability.metrics.spi.MetricReporter]: Fixed in 9e52e048eceeb99347764604a630161f78f26c33

File: tests/observability-test-utils/src/main/java/org/ballerina/testobserve/metrics/extension/MetricsUtils.java (hunk 3 above)
Comment (nadundesilva) [reply tests/observability-test-utils/src/main/java/org/ballerina/testobserve/metrics/extension/MetricsUtils.java:61]: We don't have a enum defined for metric kind at the moment. But we can introduce it though.

**Review by Kishanthan — APPROVED** [review]

**Pull request merged**